### PR TITLE
Add headings documentation

### PR DESCRIPTION
**Pull request by viktor-yakubiv**
Adds `Heading` component and a simple documentation about how CSS custom properties work with the typography.

### CODE REVIEW

**Comment by viktor-yakubiv** [comment]
I could not manage to use generic styles from `:root` and `body` so I do not really like how the headings look like. However, works everything except the font-family.

File: src/foundation/heading.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const Heading = ({
+  children,
+  className,
+  level,
+  display,
+  tag: Tag = `h${level}`,
+  ...restProps
+}) => (
+  <Tag className={`${display ? 'display' : ''} ${className}`} {...restProps}>
```
Comment (Joozty) [inline src/foundation/heading.jsx]: Would be nice to start using `classNames`

Comment (viktor-yakubiv) [reply src/foundation/heading.jsx]: I would add it later

File: src/foundation/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+## Headings
+
+### Application headings
+
+These set of headings are useful when you need it for applications,
```
Comment (Joozty) [inline src/foundation/README.md]: **This** set of headings 🙂

**Pull request merged**